flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/legacy/files/StaticFileServerHandler.java

```diff
@@ -368,7 +368,7 @@ public static void setContentTypeHeader(HttpResponse response, File file) {
     public static void checkFileValidity(File file, File rootPath, Logger logger)
             throws IOException, RestHandlerException {
         // this check must be done first to prevent probing for arbitrary files
-        if (!file.getCanonicalFile().toPath().startsWith(rootPath.toPath())) {
+        if (!file.getCanonicalFile().toPath().startsWith(rootPath.getCanonicalFile().toPath())) {
```

Comment:
Wouldn’t it be better to put this change in a separate commit?
Since it doesn’t seem logically related to the FLIP, it looks more like a hotfix.